### PR TITLE
Use `Base.TOML.Parser{Dates}` for TOML parsing w/ Dates support

### PR DESCRIPTION
**Pull request by topolarity**
Adapting to https://github.com/JuliaLang/julia/pull/54755/ and https://github.com/JuliaLang/julia/pull/55017

### CODE REVIEW

File: src/Registry/registry_instance.jl [hunk 1]
```diff
@@ -15,9 +15,8 @@ function to_tar_path_format(file::AbstractString)
 end
 
 # See loading.jl
-const TOML_CACHE = let parser = Base.TOML.Parser()
-    parser.Dates = Dates
-    Base.TOMLCache(parser, Dict{String, Dict{String, Any}}())
+const TOML_CACHE = let parser = TOML.DTParser()
```
Comment (vtjnash) [inline src/Registry/registry_instance.jl:18]: You are not allowed to parameterize the Base.TOML module with the unrelated code from TOML, as they only share identity by occasional accident as an implementation detail

Comment (topolarity) [reply src/Registry/registry_instance.jl]: Where do you see `Base.TOML` here?

If you mean `Base.TOMLCache` then we can re-export that via `TOML.Internals.TOMLCache` if need be

Comment (vtjnash) [reply src/Registry/registry_instance.jl]: Base.TOMLCache has a typeassert that you use Base.TOML and not the TOML package

**Comment by topolarity** [comment]
Probably need to wait for a new nightly first 😉

**Comment by IanButterworth** [comment]
Nightlies are updated as soon as master commits are green, not nightly. The powerpc builds are lagging but otherwise all other nightlies should be up to date.

**Comment by IanButterworth** [comment]
The 64 bit windows nightly is a day old even though there are multiple green CI runs on julia master since.. odd.

I'll give it time to resolve itself
